src/Umbraco.Web/PublishedCache/NuCache/PublishedSnapshotService.cs

```diff
@@ -1057,7 +1057,10 @@ private void RefreshMediaTypesLocked(List<int> removedIds, List<int> refreshedId
 
                 _mediaStore.UpdateContentTypes(removedIds, typesA, kits);
                 _mediaStore.UpdateContentTypes(CreateContentTypes(PublishedItemType.Media, otherIds.ToArray()).ToArray());
-                _mediaStore.NewContentTypes(CreateContentTypes(PublishedItemType.Media, newIds.ToArray()).ToArray());
+                if(newIds != null && newIds.Any())
```

Comment:
`Any()` shouldn't be used here, we are already dealing with a `List<T>` so you can use `Length`. `Any` uses Linq which means it's first going to create an iterator and use that to see if there is anything in there but we already know that with our concrete object.

Also you can see above that we have an extension method to check exactly this which we are already doing: `newIds.IsCollectionEmpty()` so this ext method should be used instead.

Reply:
Makes perfect sense, thanks. It's done now 👍